agent/consul/leader_metrics.go

```diff
@@ -43,6 +58,77 @@ func rootCAExpiryMonitor(s *Server) certExpirationMonitor {
 	}
 }
 
+func primaryCAExpiryMonitor(s *Server) certExpirationMonitor {
+	return certExpirationMonitor{
+		Key: metricsKeyMeshPrimaryCAExpiry,
+		Labels: []metrics.Label{
+			{Name: "datacenter", Value: s.config.Datacenter},
+		},
+		Logger: s.logger.Named(logging.Connect),
+		Query: func() (time.Duration, error) {
+
+			provider, _ := s.caManager.getCAProvider()
+
+			if _, ok := provider.(ca.PrimaryUsesIntermediate); !ok {
+				cert, err := getActiveIntermediate(s)
+				if err != nil {
+					return 0, err
+				}
+				return time.Until(cert.NotAfter), nil
+			}
+
+			state := s.fsm.State()
+			_, root, err := state.CARootActive(nil)
+			switch {
+			case err != nil:
+				return 0, fmt.Errorf("failed to retrieve root CA: %w", err)
+			case root == nil:
+				return 0, fmt.Errorf("no active root CA")
+			}
+
+			return time.Until(root.NotAfter), nil
```

Comment:
It might be nice to extract this as a function, and use it both here and in the existing active-roots metric, to make it more obvious that this metric reports the same value as the active-root metric in some cases.